Provide library to support postgresql as data retention site (closes #438) (#439)
eclipse-dataspaceconnector/DataSpaceConnector: common/libraries/clients/postgresql/README.md

> # commons-clients-postgresql

Reviewer: ```suggestion
# PostgreSQL Clients
```

> This library provides two Postgresql clients `SimplePostgresqlClient` and `PooledPostgresqlClient`. 

Reviewer: ```suggestion
This library provides the two Postgresql clients `SimplePostgresqlClient` and `PooledPostgresqlClient`. 
```

> Both implementation provide a unified interface to issue SQL queries towards Postgresql databases.

Reviewer: ```suggestion
Both implementations provide a unified interface for issuing SQL queries towards PostgreSQL databases.
```

Both prevent from SQL injection due to exclusive usage of `java.sql.PreparedStatement`s and yield an interface for transactional SQL queries and automatic transaction rollback while treating exceptions.
Additionally, the `PooledPostgresqlClient` performs connection pooling, which yields to better performance when the system is under load.

> Connection validation can be configured to identify ill connections. 

Reviewer: ```suggestion
Connection validation can be configured to identify broken connections. 
```
